src/AWS/Shared/Storage/DynamoDBStorage.cs

```diff
@@ -163,34 +165,143 @@ private async Task CreateTable(string tableName, List<KeySchemaElement> keys, Li
 
             try
             {
-                var response = await ddbClient.CreateTableAsync(request);
-                TableDescription description = null;
-                do
+                await ddbClient.CreateTableAsync(request);
+                TableDescription description = await TableWaitOnStatusAsync(tableName, TableStatus.CREATING, TableStatus.ACTIVE);
+                await TableUpdateTtlAsync(tableName, ttlAttributeName);
+            }
+            catch (Exception exc)
+            {
+                Logger.Error(ErrorCode.StorageProviderBase, $"Could not create table {tableName}", exc);
+                throw;
+            }
+        }
+
+        private async Task TableUpdateTtlAsync(string tableName, string ttlAttributeName)
```

Comment:
Nit: This could return `ValueTask` since it's pretty common that a ttl attribute isn't set.

Reply:
Interesting. I'll look into it.

Reply:
I Agree.